### PR TITLE
Add e2e tests for single operator multitenancy

### PR DESCRIPTION
**Pull request by super-harsh**
**Special notes for your reviewer**:

Have added new restricted credential into github secrets for multitenancy tests, which we can leverage and use into edge cases to test how operator reacts when it does not have permissions to perform stuff. 

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/internal/controllers/single_operator_multitenant_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	tc.Expect(exists).To(BeTrue())
+
+	tc.DeleteResourcesAndWait(acct, rg)
+
+}
+
+func newCredentialSecret(subscriptionID string, name, namespaceName string) *v1.Secret {
+	secretData := make(map[string][]byte)
+	// TODO: find a way to provision creds into secret
+	secretData[config.AzureClientIDVar] = []byte("")
```
Comment (super-harsh) [inline v2/internal/controllers/single_operator_multitenant_test.go:106]: Need to provision these values from somewhere

File: v2/test/single_operator_multitenant_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+	resID := genruntime.GetResourceIDOrDefault(acct)
+
+	// Make sure the StorageAccount is created successfully in Azure.
+	exists, _, err := tc.AzureClient.HeadByID(tc.Ctx, resID, string(storage.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(exists).To(BeTrue())
+
+	tc.DeleteResourcesAndWait(acct, rg)
+
```
Comment (matthchr) [inline v2/test/single_operator_multitenant_test.go:63]: minor: Extra space here

File: v2/test/single_operator_multitenant_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (matthchr) [inline v2/test/single_operator_multitenant_test.go:107]: here too

File: v2/test/single_operator_multitenant_test.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+func newCredentialSecret(subscriptionID, tenantID, name, namespaceName string) (*v1.Secret, error) {
+	secretData := make(map[string][]byte)
+	// TODO: find a way to provision creds into secret
```
Comment (matthchr) [inline v2/test/single_operator_multitenant_test.go:112]: I think this TODO is now done, via env variables?

**Review by matthchr — APPROVED** [review]
You should delete the two recording YAMLs because now you're running a live test anyway which means those files aren't being read or written by anything.
